### PR TITLE
Add gs-fgdb community module

### PR DESCRIPTION
**Pull request by tbarsballe**
Initial import from https://github.com/boundlessgeo/geoserver-exts/tree/master/fgdb (module donated to GeoServer by Boundless)

Note: The first commit is just a straight import from geoserver-exts, with FGDBDataStoreFactory removed/inlined where necessary (Using JniOGRDataStoreFactory in its place)

While it compiles, no additional validation has been done.

### CODE REVIEW

**Comment by aaime** [comment]
Thank you Torben! Just a small nitpick, if the module is a importer plugin, would it make sense to call it importer-fgdb instead?

**Comment by tbarsballe** [comment]
I didn't get the rename done prior to Boundless switching over to Planet Federal, but now that Planet Federal's got a CLA signed again, I've applied the suggested rename. I agree that importer-fgdb makes more sense as a name.

File: src/community/release/ext-importer-fgdb.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<assembly>
+  <id>importer-fgdb</id>
```
Comment (aaime) [inline src/community/release/ext-importer-fgdb.xml]: I have vague memories that a plugin module needs a name ending with -plugin or something bad (can't remember what) might happen.  At the very least all other community modules I've checked have an assembly whose name ends with -plugin.

File: src/community/importer-fgdb/src/main/resources/git.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#Generated by Git-Commit-Id-Plugin
```
Comment (aaime) [inline src/community/importer-fgdb/src/main/resources/git.properties]: Is this file really meant to be committed?

**Comment by aaime** [comment]
Added a couple more small comments, besides that everything seems to be ready for merge.

File: src/community/importer-fgdb/src/main/resources/git.properties (hunk 2 above)
Comment (tbarsballe) [reply src/community/importer-fgdb/src/main/resources/git.properties]: Nope, it isn't, leftover from the repo it was copied from. I've removed it.

File: src/community/release/ext-importer-fgdb.xml (hunk 1 above)
Comment (tbarsballe) [reply src/community/release/ext-importer-fgdb.xml]: At a guess, it may affect the name of the zipfile created by the assembly process. I've updated it.

**Comment by tbarsballe** [comment]
Fixed based on feedback. I've also fixed a couple methods that got removed by the depreciation removal. If the CI is happy with it, I'll merge.

**Pull request merged**